homeassistant/components/iaqualink/climate.py

```diff
@@ -100,9 +100,24 @@ def max_temp(self) -> int:
         return AQUALINK_TEMP_CELSIUS_HIGH
 
     @property
-    def target_temperature(self) -> float:
+    def target_temperature(self) -> float | None:
         """Return the current target temperature."""
-        return float(self.dev.state)
+        try:
+            return float(self.dev.state)
+        except ValueError:
+            _LOGGER.warning(
+                "Invalid state received for %s (%s). You likely have this device enabled in the iAqualink integration but it is not installed on the Pool Equipment",
+                self.dev.label,
+                self.dev.state,
+            )
+        except TypeError:
+            _LOGGER.warning(
+                "Invalid state received for %s (%s). You likely have this device enabled in the iAqualink integration but it is not installed on the Pool Equipment",
+                self.dev.label,
+                self.dev.state,
+            )
```

Comment:
I'm not sure why these warnings are needed.
Why keep warning the user about something they can't fix? (please note, this warning will spam the logs).

Shouldn't this entity be unavailable in these cases? (should it be created even?).

Reply:
They can fix it if they disable the device that the integration created. That's why I added this warning to the log. If the message is not clear I can try to improve it.

Reply:
@carlos-sarmiento  - Is there a way to not create those entities like @frenck alluded to ?  Perhaps that will get this PR approved and merged